fhir-bucket/src/main/java/com/ibm/fhir/bucket/api/IResourceHandler.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.bucket.api;
+
+
+/**
+ * Interface for handling the processing of resources read from COS
+ */
+public interface IResourceHandler {
+
+    /**
+     * Add the resource entry to the thread-pool for processing, subject to the
+     * rate limiting we have to make sure memory consumption is kept in check
+     * @param entry
+     * @return
+     */
+    public boolean process(ResourceEntry entry);
```

Comment:
While I think I get the intent, what is the difference with fhir-bucket/src/main/java/com/ibm/fhir/bucket/api/IResourceEntryProcessor.java

Reply:
They are used in different contexts and I initially had different signatures in one of my iterations. They are used in different contexts, so I'm not planning on refactoring at this point. One returns a boolean whereas the other doesn't.